containers/runtime/xds/cmd/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/test/v3"
+
+	"github.com/grpc/test-infra/containers/runtime/xds"
+	config "github.com/grpc/test-infra/containers/runtime/xds/config"
+
+	_ "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/router/v3"
+)
+
+func main() {
+
+	resource := config.TestResource{}
+
+	var nodeID string
+	var xdsServerPort uint
+	var testListenerPort uint
+	var defaultConfigPath string
+	var userSuppliedConfigPath string
+	var endpointUpdaterPort uint
+
+	// The port that this xDS server listens on
+	flag.UintVar(&xdsServerPort, "xdsServerPort", 18000, "xDS management server port, this is where Envoy/gRPC client gets update")
+
+	// The port that endpoint updater server listens on
+	flag.UintVar(&endpointUpdaterPort, "endpointUpdaterPort", 18005, "endpointUpdater server port, this is where endpoint updater gets the IP and port for test servers")
+
+	// Tell Envoy/xDS client to use this Node ID, it is important to match what provided in the bootstrap files
+	flag.StringVar(&nodeID, "nodeID", "test_id", "Node ID")
+
+	// Default configuration path, the path is relative path using ./containers/runtime/xds
+	flag.StringVar(&defaultConfigPath, "d", "containers/runtime/xds/config/default_config.json", "The path of default configuration file, the path is relative path the root of test-infra repo")
+
+	// User supplied configuration path, the path is relative path using ./containers/runtime/xds
+	flag.StringVar(&userSuppliedConfigPath, "u", "", "The path of user supplied configuration file, the path is relative path the root of test-infra repo")
+
+	// Tne cluster name for Envoy obtain configuration from, should match the cluster name in the bootstrap file.
+	flag.StringVar(&resource.XDSServerClusterName, "c", "xds_cluster", "Tne cluster name for Envoy to obtain configuration, should match the cluster name in the bootstrap file")
```

Comment:
I don't think I understand what's going on here. Why is there a static cluster in the Envoy bootstrap? Also, this flag doesn't seem to be used anywhere.

Reply:
Ah, after looking at https://github.com/grpc/test-infra/pull/242, I see why you need this. But I'm not sure why the xDS server needs to know this cluster name? Is it just because you need to avoid a collision in cluster names?

Reply:
If we use ads{} for LDS/RDS and CDS/EDS we would not need the cluster name, but we used the cluster name in RDS configSource: ["clusterName": "xds_cluster"](https://github.com/grpc/test-infra/blob/10f8de6bcbaa9377267dff804d9c470ae0da4281/containers/runtime/xds/config/default_config.json#L115), I will worker further on using ads for RDS for Envoy, if not possible I will double check to see if this field is required.  

My understanding is APIserver is only used by gRPC proxyless clients and may not be used by Envoy, so there are two listeners pointing to the same cluster/endpoint. Preferably the two listeners should be more identical, the api_listener (proxyless clients) is using ads, I will try to make the Envoy listener able to use ads{} as well.

Reply:
I have tried to use ads for RSD config, did not work properly, I think it will take a bit to figure out. Before How important would it be to use the same type of config source (ads vs rds) for Envoy and gRPC xds client?

Reply:
@gnossen I have tested that we are able to use ads for fetching all resources, if we are sure we always want to use ads we may remove this field. Do you think we may want to keep this field in case in the future someone want to fetch rds or eds separately? Same thing goes with the servers, if we are sure we always use ads we can register the ads server only.

Reply:
Let's stick with ADS. I find it highly unlikely that someone will want to configure that option in these benchmarks.

Reply:
I will update the code to only use ADS.